llvm/docs/LangRef.rst

```diff
@@ -12621,6 +12621,9 @@ have no side effects, and must not capture the value of the pointer.
 If the source is :ref:`poison <poisonvalues>`, the result is
 :ref:`poison <poisonvalues>`.
 
+If the source is not :ref:`poison <poisonvalues>`, and the result pointer is
+non-dereferenceable, the result is :ref:`poison <poisonvalues>`.
```

Comment:
This is incorrect, not every addrspacecast of a non-dereferenceable pointer should automatically result in poison (e.g. the pointer to the end of an object is non-dereferenceable, but should certainly not turn into poison).

This should instead say something along the lines of:

> Which address space casts are supported depends on the target. Unsupported address space casts return a poison value.

Reply:
done in 3128b338492a977b35b7c39bc770e85a703ca8d0, thanks @nikic